test/cctest/test_url.cc

```diff
@@ -137,25 +137,31 @@ TEST_F(URLTest, FromFilePath) {
   file_url = URL::FromFilePath("C:\\Program Files\\");
   EXPECT_EQ("file:", file_url.protocol());
   EXPECT_EQ("//C:/Program%20Files/", file_url.path());
+  EXPECT_EQ("file:///C:/Program%20Files/", file_url.href());
 
   file_url = URL::FromFilePath("C:\\a\\b\\c");
   EXPECT_EQ("file:", file_url.protocol());
   EXPECT_EQ("//C:/a/b/c", file_url.path());
+  EXPECT_EQ("file:///C:/a/b/c", file_url.href());
 
   file_url = URL::FromFilePath("b:\\a\\%%.js");
   EXPECT_EQ("file:", file_url.protocol());
   EXPECT_EQ("//b:/a/%25%25.js", file_url.path());
+  EXPECT_EQ("file:///b:/a/%25%25.js", file_url.href());
 #else
   file_url = URL::FromFilePath("/");
   EXPECT_EQ("file:", file_url.protocol());
   EXPECT_EQ("//", file_url.path());
+  EXPECT_EQ("file:///", file_url.href());
 
   file_url = URL::FromFilePath("/a/b/c");
   EXPECT_EQ("file:", file_url.protocol());
   EXPECT_EQ("//a/b/c", file_url.path());
+  EXPECT_EQ("file:///a/b/c", file_url.href());
 
   file_url = URL::FromFilePath("/a/%%.js");
   EXPECT_EQ("file:", file_url.protocol());
   EXPECT_EQ("//a/%25%25.js", file_url.path());
+  EXPECT_EQ("file:///a/%25%25.js", file_url.href());
 #endif
 }
```

Comment:
Would it be good to add a test case for a URL with username/password so we are testing that part of the implementation as well?

Reply:
Will do, thank you! Once we ship this, we can start migrating `SerializeURL` into URL to have full tests. For not, the main scope is around the `inspector_agent`.